File: .tmp_py4013/batch_11_dump/3264_final_array_state_after_k_multiplication_operations_i.js
// LeetCode 3264 - Final Array State After K Multiplication Operations I
// https://leetcode.com/problems/final-array-state-after-k-multiplication-operations-i/

function MinHeap(cmp) {
    this.a = [];
    this.cmp = cmp || ((x, y) => x - y);
}
MinHeap.prototype._up = function(i) {
    const a = this.a, cmp = this.cmp;
    while (i > 0) {
        const p = (i - 1) >> 1;
        if (cmp(a[i], a[p]) >= 0) break;
        [a[i], a[p]] = [a[p], a[i]];
        i = p;
    }
};
MinHeap.prototype._down = function(i) {
    const a = this.a, cmp = this.cmp, n = a.length;
    while (true) {
        let s = i, l = i * 2 + 1, r = l + 1;
        if (l < n && cmp(a[l], a[s]) < 0) s = l;
        if (r < n && cmp(a[r], a[s]) < 0) s = r;
        if (s === i) break;
        [a[i], a[s]] = [a[s], a[i]];
        i = s;
    }
};
MinHeap.prototype.push = function(x) { this.a.push(x); this._up(this.a.length - 1); };
MinHeap.prototype.pop = function() {
    const a = this.a;
    if (!a.length) return undefined;
    const top = a[0], last = a.pop();
    if (a.length) { a[0] = last; this._down(0); }
    return top;
};
MinHeap.prototype.peek = function() { return this.a[0]; };
MinHeap.prototype.size = function() { return this.a.length; };

var getFinalState = function(nums, k, multiplier) {
    const h = new MinHeap((a, b) => a[0] !== b[0] ? a[0] - b[0] : a[1] - b[1]);
    for (let i = 0; i < nums.length; i++) h.push([nums[i], i]);
    for (let t = 0; t < k; t++) {
        const cur = h.pop();
        const v = cur[0] * multiplier, i = cur[1];
        nums[i] = v;
        h.push([v, i]);
    }
    return nums;
};

File: .tmp_rs4013/rust_check15/3264_final_array_state_after_k_multiplication_operations_i.rs
struct Solution;
// LeetCode 3264 - Final Array State After K Multiplication Operations I
// https://leetcode.com/problems/final-array-state-after-k-multiplication-operations-i/

use std::cmp::Reverse;
use std::collections::BinaryHeap;

impl Solution {
    pub fn get_final_state(mut nums: Vec<i32>, k: i32, multiplier: i32) -> Vec<i32> {
        let mut h: BinaryHeap<Reverse<(i32, usize)>> = BinaryHeap::new();
        for (i, &v) in nums.iter().enumerate() {
            h.push(Reverse((v, i)));
        }
        for _ in 0..k {
            if let Some(Reverse((v, i))) = h.pop() {
                let nv = v * multiplier;
                nums[i] = nv;
                h.push(Reverse((nv, i)));
            }
        }
        nums
    }
}

fn main() {}
